pandas/tests/dtypes/cast/test_promote.py

```diff
@@ -151,7 +151,17 @@ def _assert_match(result_fill_value, expected_fill_value):
     # GH#23982/25425 require the same type in addition to equality/NA-ness
     res_type = type(result_fill_value)
     ex_type = type(expected_fill_value)
-    assert res_type == ex_type
+    if res_type.__name__ == "uint64":
+        # No idea why, but these (sometimes) do not compare as equal
+        assert ex_type.__name__ == "uint64"
+    elif res_type.__name__ == "ulonglong":
```

Comment:
@h-vetinari did you find a nicer way to handle these?

Reply:
No, I didn't have to deal with `ulonglong` (pretty sure that's platform-specific).